daml-foundations/daml-ghc/daml-prim-src/GHC/Num.daml

```diff
@@ -108,10 +124,36 @@ instance Additive Decimal where
 instance Multiplicative Decimal where
     (*) = primitive @"BEMulDecimal"
     munit = 1.0
+    x ^ n
+        | n == 0 = 1.0
+        | n == 2 = x * x
+        | n < 0 = 1.0 / x ^ (negate n)
+        | n % 2 == 0 = (x ^ (n / 2)) ^ 2
+        | otherwise = x * x ^ (n - 1)
 
 instance Number Decimal where
     -- no methods
 
 instance Signed Decimal where
     signum x = if x == 0.0 then 0.0 else if x <= 0.0 then (-1.0) else 1.0
     abs x = if x <= 0.0 then negate x else x
+
+class (Multiplicative a) => Fractional a where
+    -- | `x / y` divides `x` by `y`.
+    (/)                 : a -> a -> a
+
+    -- | Calculate the reciprocal: `recip x` is `1/x`.
+    recip               : a -> a
+    x / y               = x * recip y
+
+instance Fractional Decimal where
+    (/) = primitive @"BEDivDecimal"
+    recip x = 1.0 / x
+
+instance Fractional Int where
+    (/) = primitive @"BEDivInt64"
+    recip x = 1 / x
+
+-- | `x % y` calculates the remainder of `x` by `y`
+(%) : Int -> Int -> Int
+(%) = primitive @"BEModInt64"
```

Comment:
Newline at the end of the file please - I recommend configuring your editor properly.